src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -218,13 +218,117 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
                         }
                         else
                         {
+#if UNIX
+                            // On UNIX systems, we expand arguments containing wildcard expressions against
+                            // the file system just like bash, etc.
+                            if (System.Management.Automation.WildcardPattern.ContainsWildcardCharacters(arg))
+                            {
+                                // See if the current working directory is a filesystem provider location
+                                // We won't do the expansion if it isn't since native commands can only access the file system.
+                                System.IO.DirectoryInfo cwdinfo = null;
+                                string cwd = null;
+                                var pscwd = Context.EngineSessionState.InvokeProvider.Item.Get(".");
+                                Diagnostics.Assert(pscwd != null, "pscwd should never be null");
+                                if (pscwd.Count == 1)
+                                {
+                                    cwdinfo = pscwd[0].BaseObject as System.IO.DirectoryInfo;
+                                }
+                                if (cwdinfo != null)
+                                {
+                                    cwd = cwdinfo.FullName;
+                                }
```

Comment:
How about using `Context.EngineSessionState.CurrentLocation`? Then you can check `if (pwd.Provider.Name == FileSystemProvider.ProviderName)` to see if it's a filesystem location, and `pwd.ProviderPath` to get the full path.